src/components/Post/PostInFeed.tsx

```diff
@@ -1,50 +1,62 @@
 import { Box, Text, Skeleton, HStack, FavouriteIcon, Pressable } from 'native-base';
 import { Post } from '../../xplat/types/post';
 import { useEffect, useState } from 'react';
+import { useQuery } from 'react-query';
 import '../css/feed.css';
 import { User } from '../../xplat/types/user';
 import AuthorHandle from '../User/AuthorHandle';
 import PostImages from './PostImages';
+import { buildPostFetcher } from '../../utils/queries';
 
 const PostInFeed = ({ post, _passAuthor }: { post: Post, _passAuthor?: User | undefined }) => {
     const [body, setBody] = useState('');
     const [author, setAuthor] = useState<User>();
     const [imageCount, setImageCount] = useState(0);
     const [imageURLs, setImageURLs] = useState<Array<string | undefined>>();
-    const [isLoaded, setIsLoaded] = useState(false);
+    const [videoURL, setVideoURL] = useState<string | undefined>();
+    const [videoThumbnail, setVideoThumbnail] = useState<string | undefined>();
+    const [timestamp, setTimestamp] = useState<Date>();
     const [likes, setLikes] = useState(0);
-
-    async function resolvePost (): Promise<void> {
-        await post.getData();
-
-        void post.getTextContent().then(setBody);
-        void post.getImageContentUrls().then(setImageURLs);
-        void post.getImageCount().then(setImageCount);
-        if (_passAuthor)
-            setAuthor(_passAuthor);
-        else
-            void post.getAuthor().then(setAuthor);
-        void post.getLikes().then((likesArray) => { setLikes(likesArray.length); });
-        setIsLoaded(true);
-    }
+    const { isLoading, error, data } = useQuery(
+        post.docRef!.id, 
+        buildPostFetcher(post)   
+    );
 
     useEffect(() => {
-        void resolvePost();
-    }, []);
+        if (data && !error)
+        {
+            setBody(data.body);
+            setImageCount(data.imageCount);
+            setLikes(data.likes);
+            setAuthor(data.author);
+            setImageURLs(data.imageURLs);
+            setImageCount(data.imageCount);
+            setTimestamp(data.timestamp);
+            if (data.videoContent)
+            {
+                setVideoURL(data.videoContent.videoURL);
+                setVideoThumbnail(data.videoContent.thumbnailURL);
+            }
+        }
+    }, [data]);
```

Comment:
Although valid, this introduces data duplication. Can we use `data.body` and it's siblings directly rather than moving them into state?

Reply:
I considered that but I thought keeping differences between pre-Query and now at a minimum would make the change more understandable. For state's though, would the only use-cases for them with Query be for things like our QueryCursors?

Reply:
> For state's though, would the only use-cases for them with Query be for things like our QueryCursors?
Yeah agreed, which is why I think we should get rid of all of this redundant state. Although theoretically more changes, I think removing this state makes the end product less confusing (and more efficient).

Reply:
I was also wondering this. So lets say we initially render the object, it calls the query, it gets the doc from the DB, then re-renders with the updated static data. If we re-render this object again sometime in the future, does the query get made again? And if so, is the data cached? If it is, does making the query again cause another re-render even if the data is cached?

Reply:
The query will not get made again, as it is cached. Yes, getting the data from cache re-renders.

Reply:
> The query will not get made again, as it is cached. Yes, getting the data from cache re-renders.

okay nice. And I am assuming this is a data cache is a fairly quick re-render?